### PR TITLE
feat(retrieval): direct-answer wiring layer (#518 slice 3/8)

### PR DESCRIPTION
**Pull request by joshuaswarren**
Third of eight slices for #518. Binds the pure direct-answer eligibility decision (slice 2, merged at \`04c23cbb\`) to the data sources needed to build candidates. Dead code — no call site yet, safe to merge and release alone. The next slice adds exactly one call site in \`recallInternal()\` before QMD, plus the \`LastRecallSnapshot\` field that carries \`RecallTierExplain\` out.

## Summary

- New \`packages/remnic-core/src/direct-answer-wiring.ts\` with \`tryDirectAnswer(input)\`.
- New \`direct-answer-wiring.test.ts\` — 11 tests, all green. \`tsc --noEmit\` clean.
- Combined slice-1/2/3 core tests: 71/71 pass.

## Design

- **Source decoupling.** Orchestrator internals stay out of this layer — callers inject a \`DirectAnswerSources\` interface (\`listCandidateMemories\`, \`trustZoneFor\`, \`importanceFor\`, \`taxonomy\`). Tests use mock sources, production binding is slice 3b's job.
- **Zero-I/O disabled path.** When \`config.recallDirectAnswerEnabled === false\`, the function returns immediately with reason \`"disabled"\` without calling any accessor. This is the documented default and the contract test enforces it.
- **Pre-filter then delegate.** When enabled, cheaply drops non-trusted-zone memories before taxonomy resolution, and skips importance resolution for ineligible taxonomy buckets. The full gate ladder still runs inside \`isDirectAnswerEligible\` — the pre-filter is purely an I/O optimization, not a correctness boundary.
- **Abort-aware.** \`abortSignal\` is checked before each candidate resolution so recall cancellation halts the walk.

## CLAUDE.md rules honored

- **Rule 30** — \`enabled: false\` gate tested explicitly, not-touching-sources contract asserted.
- **Rule 42** — namespace flows through all access; the wiring does not cross namespaces on its own.
- **Rule 51** — all inputs validated upstream by slice 1 (\`parseConfig\`) and slice 2 (eligibility gates).

## Out of scope for this slice

- Call site in \`recallInternal\` (slice 3b — a tiny diff)
- \`LastRecallSnapshot.tierExplain\` extension (slice 3b)
- CLI / HTTP / MCP surfaces (slices 4–6)
- Bench + docs (slices 7–8)

## Test plan

- [x] \`packages/remnic-core/src/direct-answer-wiring.test.ts\` — 11/11 pass
- [x] \`packages/remnic-core/src/direct-answer.test.ts\` — 21/21 pass (no regressions)
- [x] \`packages/remnic-core/src/config.test.ts\` — 39/39 pass (no regressions)
- [x] \`tsc --noEmit\` clean
- [x] Module is genuinely dead code: \`grep -rn 'direct-answer-wiring\\|tryDirectAnswer' src packages\` returns only the module and its test

Part of #518.

---

> [!NOTE]
> **Low Risk**
> Adds a new, currently unreferenced wiring module plus unit tests; it doesn’t change runtime behavior until a call site is introduced. Risk is mainly around future integration correctness (abort handling and prefiltering) rather than regressions today.
> 
> **Overview**
> Introduces a new `tryDirectAnswer` wiring layer that materializes direct-answer candidates from injected sources (candidate listing, trust zone lookup, taxonomy resolution, and importance scoring) and then delegates the final verdict to `isDirectAnswerEligible`.
> 
> The wiring adds **zero-I/O short-circuits** for `recallDirectAnswerEnabled=false` and for queries that normalize to no searchable tokens, performs cheap prefiltering (drop non-`trusted` and ineligible taxonomy buckets before computing importance), propagates `namespace`/`queryEntityRefs`, and is **abort-signal aware** (throws `AbortError` on cancellation).
> 
> Adds a comprehensive test suite covering the short-circuit contracts, prefiltering behavior, ambiguity deferral, namespace/queryEntityRefs plumbing, and abort edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1053733b43ba001cc00162bdfea6c188491f7c27. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/remnic-core/src/direct-answer-wiring.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      config: eligibilityConfig,
+      queryEntityRefs,
+    });
+  }
+
+  const memories = await sources.listCandidateMemories({ namespace, abortSignal });
+  const candidates: DirectAnswerCandidate[] = [];
+
+  for (const memory of memories) {
+    if (abortSignal?.aborted) break;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/direct-answer-wiring.ts]: Fixed in e90d112a — tryDirectAnswer now throws an AbortError ({name: "AbortError"}) on observed abort, matching orchestrator.ts's abort convention. Added explicit checks before listCandidateMemories, after list resolution, and at the top of each loop iteration. Test updated to assert.rejects on the AbortError name, plus a new case for already-aborted-before-entry. No partial verdict can now escape.

File: packages/remnic-core/src/direct-answer-wiring.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  const candidates: DirectAnswerCandidate[] = [];
+
+  for (const memory of memories) {
+    // Throw rather than returning a partial verdict — a mid-loop abort
+    // that left competing candidates unprocessed could otherwise surface
+    // a spurious "eligible" result that the ambiguity gate never got a
+    // chance to reject.
+    throwIfAborted(abortSignal);
+
+    const trustZone = await sources.trustZoneFor(memory.frontmatter.id);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/direct-answer-wiring.ts:143]: Fixed in 5f5cf985. Added `throwIfAborted(abortSignal)` immediately after `await sources.trustZoneFor(...)` AND after the for-loop exits, so an abort landing during the trust-zone I/O on the last (or only) memory still surfaces as AbortError rather than falling through to the eligibility gate. Two regression tests cover exactly those cases — abort during trustZoneFor on the only memory, and abort during trustZoneFor on the last of several memories — both assert `err.name === 'AbortError'`. 14/14 wiring tests pass.

File: packages/remnic-core/src/direct-answer-wiring.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  });
+}
+
+function throwIfAborted(signal?: AbortSignal): void {
+  if (signal?.aborted) {
+    const err = new Error("direct-answer wiring aborted");
+    Object.defineProperty(err, "name", { value: "AbortError" });
+    throw err;
+  }
+}
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/direct-answer-wiring.ts:187]: Good catch — this helper is indeed now duplicated across `direct-answer-wiring.ts`, `harmonic-retrieval.ts`, and `qmd.ts` (plus a similar private pair in `orchestrator.ts`). Since this PR's job is adding the direct-answer wiring layer, I'd rather not expand its blast radius by touching two hot-path retrieval files for a pure refactor — that belongs in its own small PR. Filed as follow-up work: extract `abort-error.ts` with `abortError` / `isAbortError` / `throwIfAborted` and route all four call sites through it. Happy to take that next if it's not on anyone else's radar.

**Comment by joshuaswarren** [comment]
@cursor review

The remaining `throwIfAborted` duplication across `qmd.ts`, `harmonic-retrieval.ts`, and this new module is deferred to a follow-up refactor PR to keep slice 3's blast radius limited to the direct-answer wiring. Both prior P1/P2 findings (abort partial verdict, abort after trustZoneFor await) are fixed and covered by regression tests.

File: packages/remnic-core/src/direct-answer-wiring.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      query,
+      candidates: [],
+      config: eligibilityConfig,
+      queryEntityRefs,
+    });
+  }
+
+  throwIfAborted(abortSignal);
+  const memories = await sources.listCandidateMemories({ namespace, abortSignal });
+  throwIfAborted(abortSignal);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/direct-answer-wiring.ts:131]: Fixed in 1053733b — tryDirectAnswer now normalizes the query up-front via `normalizeRecallTokens()` and, when the token list is empty, returns the eligibility verdict with zero candidates before any accessor call. New regression test seeds the mock source with a real memory + trust-zone + importance, then fires a punctuation-only query and asserts `reason === 'empty-query'` AND all three source-call counters are 0.
